src/screencast/wlr_screencast.c

```diff
@@ -313,6 +313,205 @@ static void wlr_init_xdg_outputs(struct xdpw_screencast_context *ctx) {
 	}
 }
 
+static bool exec_chooser(char *cmd, bool readin, int p1[2], int p2[2]) {
+	logprint(TRACE, "exec chooser called: cmd %s, readin %d, pipe1 (%d,%d), pipe2 (%d,%d)", cmd, readin, p1[0], p1[1], p2[0], p2[1]);
+	pid_t pid = fork();
+
+	if (pid < 0) {
+		perror("fork");
+		close(p1[0]);
+		close(p2[1]);
+		return false;
+	} else if (pid == 0) {
+		close(p2[0]);
+
+		if (readin) {
+			dup2(p1[0], STDIN_FILENO);
+		}
+		dup2(p2[1], STDOUT_FILENO);
+		close(p1[0]);
+		close(p2[1]);
+
+		int err;
+		err = execl("/bin/sh", "/bin/sh", "-c", cmd, (void *)NULL);
+
+		if (err == -1) {
+			perror("execl");
+		}
+		_exit(127);
+	}
+
+	close(p1[0]);
+	close(p2[1]);
+
+	int status;
+	wait(&status);
+	if (WIFEXITED(status)) {
+		if (WEXITSTATUS(status) == 127) {
+			return false;
+		} else {
+			return true;
+		}
+	}
+	return false;
+}
+
+static bool wlr_output_chooser(struct xdpw_output_chooser *chooser,
+		struct wl_list *output_list, struct xdpw_wlr_output **output) {
+	logprint(DEBUG, "wlroots: output chooser called");
+	struct xdpw_wlr_output *out, *tmp;
+	size_t namelength = 0;
+	char *name = NULL;
+	bool readin = false;
+	FILE *f = NULL;
+
+	int p1[2]; //p -> c
+	int p2[2]; //c -> p
+
+	if (pipe(p1) == -1) {
```

Comment:
If we aren't sending a list of outputs on stdin, there's no need to create a pipe here. We can just set the pipe FDs to `{ -1, -1 }` to indicate that there's no input.

This would remove the need for the `readin` variable.

Reply:
what if we remove the readin variable, connect stdin every time with our pipe and just close it, if we don't want to send anything?

Reply:
We can just set the chooser input FD to -1, no need to create a pipe at all.

Reply:
Sure, my thought creating both pipes would make the cases with sth on stdin and without more similar. Without the second pipe we need two switch cases for the chooser type: one before spawn_chooser, to create only the needed pipes and one after to write to stdin accordingly, which would make it a little bit more complicated.

Reply:
I see, makes sense. I'm fine with that.